backend/uclapi/workspaces/occupeye.py

```diff
@@ -0,0 +1,741 @@
+import json
+import os
+from base64 import b64encode
+from collections import OrderedDict
+from time import time as time_now
+import redis
+import requests
+
+from django.conf import settings
+
+
+class BadOccupEyeRequest(Exception):
+    """
+    Custom exception for when any CadCap API request fails.
+    This should only be raised by endpoints that could contain
+    user-entered data, so that we can give them an error telling
+    them that the data they gave us is bad.
+    """
+    pass
+
+
+class OccupEyeApi():
+    """
+    Python API for the Cad-Capture OccupEye backend.
+    Data is cached as much as possible in Redis for performance.
+    """
+    # Keep general survey data around in the cache for a day
+    SURVEY_TTL = 86400
+
+    # Keep image data in the cache for two days
+    IMAGE_TTL = 172800
+
+    # Keep sensor data in the cache for a minute
+    SENSOR_STATUS_TTL = 60
+
+    def __init__(self, test_mode=False):
+        self.r = redis.StrictRedis(

```

Comment:
`self.redis` ?

Reply:
I called it self.r for brevity because there's so much use of Redis throughout the code file and I didn't want self.redis and redis to both be used to avoid confusion.

Reply:
In my opinion `self.redi` is much cleaner. 

I definitely find distinguishing between `redis` and `self.redis` easier than the cognitive load of remembering what `self.r` is